src/cluster.rs

```diff
@@ -200,27 +200,27 @@ impl Clustering {
     }
 
     /**
-     * Retrieve the objective values achieved from the clustering process.
+     * Retrieve the stats achieved from the clustering process.
      * Returns as many values as the number of iterations made.
      */
-    pub fn objectives(&self) -> Result<&[f32]> {
+    pub fn iteration_stats(&self) -> Result<&[FaissClusteringIterationStats]> {
```

Comment:
I feel that it's best to provide a more idiomatic abstraction for the clustering iteration stats, instead of plainly using what's in faiss-sys directly.
Two choices here:

- Create a `pub struct ClusteringIterStats(FaissClusteringIterationStats)` with a transparent repr and getter methods. This makes it so that the high level API does not depend on internal layouts.
- Alternatively, we can just keep it simple and provide a nicer name through an alias: `pub type ClusteringIterStats = FaissClusteringIterationStats;`

Would you mind choosing one of the approaches and implementing it? If you need more assistance, please let me know.

Reply:
I prefer the `repr(transparent)` way, but `FaissClusteringIterationStats(_H)` is an opaque type which size is 0.

```rust
#[repr(C)]
pub struct FaissClusteringIterationStats_H {
    _unused: [u8; 0],
}
```
 `repr(transparent)` require non-zero-size T,therefore I cannot use a tuple struct wrapper.

```rust
#[repr(transprent)]
pub struct ClusteringIterStats(FaissClusteringIterationStats)
// got error[E0690]: transparent struct needs exactly one non-zero-sized field, but has 0
```

Since `FaissClusteringIterationStats(_H)` is an opaque type, I mannually create a `ClusteringIterationStats` same as `FaissClusteringIterationStats(_H)`.

Reply:
see the new commit